---
title: "Group 01 project: Analysis of Gene Expression in Parkinsson Disease"
author: 
  - Rune Daucke, runda
  - David Faurdal, s144523
  - Luisa Weisch,
format:
  revealjs:
    self-contained: true
    code-line-numbers: true
    smaller: true
    scroll: true
    theme: moon
    transition: slide
editor: visual
---

## Introduction

Gene expression data (Affymetrix platform).

Collected from the blood of: - 20 healthy patients - 40 patients diagnosed with sporadic Parkinsons disease

## Methods <!-- Luisa -->

## Cleaning and Filtering Dataset

::: columns
::: {.column width="50%"}
-   Arranging samples by condition

-   Filter out AFFX related spikes

-   Reshaping into tidy format

-   Mapping probe IDs to genes using Affymetrix database

-   Sorting samples 
:::

::: {.column width="50%"}
::: {style="background-color: white; padding: 10px; display:"}
![tidy_format](images/tidy_format.png){width="80%" height="80%"}
:::
:::
:::

## Augmentation
-   Mean expression of sets of potential biomarkers
    -   Neurodegenerative-associated genes
    -   Pro-Inflammation genes
```{r}
#| echo: true
#| eval: false
# inflammation index
inflam_list <- c("IL1rn", "TNF", "Saa3", "Emr1", "Adam8", "Itgam")

present_genes <- find_genes(inflam_list, all_genes)
df_wide$inflam_mean <- rowMeans(df_wide[present_genes], na.rm = TRUE)
```


-   

    -   PCA

-   

    -   Differential Expression

-   

    -   Differential Expression of known biomarker gene

## Results - PCA <!-- Rune -->

::: {style="background-color: white; padding: 10px; display:"}
![PCA](../results/05_PCA_plot.png){width="80%" height="80%"}
:::

## Results - Diff. Expression <!-- David -->

::: {style="background-color: white; padding: 10px; display:"}
![volcano](../results/06_Volcano_plot_gene_exp.png){width="80%"}
:::

## Results - Diff. Expression of known biomarkers <!-- Luisa -->

### Neurodegenerative-associated genes

::: columns
::: {.column width="50%"}
:::{style="background-color: white; padding: 10px; display:"}
![neurodegenerative_density](../results/07_neurodeg_density.png){width="80%"}
:::
:::


::: {.column width="50%"}
:::{style="background-color: white; padding: 10px; display:"}
![neurodegenerative_boxplot](../results/07_neurodeg_boxplot.png){width="80%"}
:::
:::
:::

## Results - Diff. Expression of known biomarkers <!-- Luisa -->

### Pro-inflammatory genes

::: columns
::: {.column width="50%"}
::: {style="background-color: white; padding: 10px; display:"}
![inflammatory_density](../results/07_inflammatory_density.png){width="80%"}
:::
:::

::: {.column width="50%"}
::: {style="background-color: white; padding: 10px; display:"}
![inflammatory_boxplot](../results/07_inflammatory_boxplot.png){width="80%"}
:::
:::
:::

## Kegg enrichment

::: {style="background-color: white; padding: 10px; display:"}
![kegg_enrichment](../results/08_kegg_enrichment.png){width="80%" height="80%"}
:::

## Kegg pathway

::: {style="background-color: white; display:"}
![kegg_pathway](../results/08_hsa00140.pathview.png){width="50%" height="50%"}
:::

## Conclusion <!-- Rune -->

-   No clustering / Separtion along the PC's
-   No significant differentially expressed genes
-   What to do next?
    -   With patient metadata, clustering might present itself
    -   Non-linear tendencies. Performing UMAP or t-SNE could provide clustering
    -   ANN, KNN logistic modelling could reveal expression patterns
    -   Redo the experiment
    -   Perhaps blood transcriptomics on PD patient, isn't the way to go, though biomarker discovery would be ideal.
